### ISSUE TITLE
Use toEmit with async code

### ISSUE THREAD

**Issue by DanweDE** [kefirjs/kefir-test-utils#49]
It would be great if `toEmit` (i.e. `createTestHelpers(Kefir).watch`) would allow asserting that an observable emits, also asynchronously. For example:

```js
const seq = Kefir.sequentially(100, [1, 2]).toProperty(() => 0)

await expect(seq).toEmit([
  value(0),
  value(1),
  value(2, { current: true }),
  end({ current: true }),
], async () => {
  await Promise((r) => setTimeout(r, 1000));
})
```

**Comment by mAAdhaTTah** [comment]
`toEmitInTime` is used for observables that emit values over time, but we use it with `lolex` to mock time. So the above example would be as follows:

```js
const seq = Kefir.sequentially(100, [1, 2]).toProperty(() => 0)

expect(seq).toEmit([
  [0, value(0, { current: true })],
  [100, value(1)],
  [200, value(2)],
  [200, end()],
], tick => {
  tick(200);
});
```

**Comment by DanweDE** [comment]
Yes, I am aware of that. In theory that is sufficient, in practice it might not work when testing code that also relies on promises so I still think this suggestion would make a lot of sense for real-world testing cases.
The second use-case where this would be helpful would be if you're testing and you don't care about the exact time but just the order something emitted.

**Comment by mAAdhaTTah** [comment]
Ok, will reopen to investigate, but not sure when I'll get to it. Open to a PR though if you're interested!

**Comment by mAAdhaTTah** [comment]
Two things:

1. I don't think this actually requires any changes in KTU. `watch` returns the `log` array and an `unwatch` function, which in `toEmit` is called synchronously. I'm going to transfer this issue to the `jest-kefir` repo.
2. Looking into this now, it appears this would be a breaking change, because `expect(obs).toEmit([...])` would now need to be `await`'d. Instead of modifying `toEmit` to allow `async` functions, what do you think about a new matcher called `toEmitAsync` that expects an `async` function?
